aws/src/main/java/org/apache/iceberg/aws/s3/S3RequestUtil.java

```diff
@@ -149,4 +151,10 @@ static void configurePermission(
       Function<ObjectCannedACL, S3Request.Builder> aclSetter) {
     aclSetter.apply(s3FileIOProperties.acl());
   }
+
+  // TODO Remove me once all of the S3-compatible storage support strong integrity checks
+  @SuppressWarnings("deprecation")
+  static AwsRequestOverrideConfiguration disableStrongIntegrityChecksums() {
+    return AwsRequestOverrideConfiguration.builder().signer(AwsS3V4Signer.create()).build();
```

Comment:
I'm not sure this is the right way to disable it? Feels like this is just a side effect of setting a new signer but I don't think we want that.

https://github.com/aws/aws-sdk-java-v2/issues/5848#issuecomment-2627909526

https://github.com/aws/aws-sdk-java-v2/blob/468d10445765c26deeaa451d34c446b6f0616f6b/services/s3/src/it/java/software/amazon/awssdk/services/s3/checksum/HttpChecksumIntegrationTest.java#L428

Reply:
Oh I see, is the issue in compatibility is that older versions of Minio/3rd party object storage solutions expect `Content-MD5` and in the new SDK we are not sending that and so the service rejects the request? Still feels like there should be a different way to *force* setting MD5

Reply:
Hmm, only setting `WHEN_REQUIRED` for checksum calculation/validation doesn't resolve as far as I tested. Calculating MD5 for PutObjectRequest looks feasible, but I'm not sure how to do it for DeleteObjectsRequest.

Reply:
We have the same issue on the PyIceberg: https://github.com/apache/iceberg-python/issues/1546 How about making this configurable?

Reply:
@ebyhr no. The only workaround that I have found is to force previous signature and it works against Dell ECS, old Minio versions and Ozone. I haven't tested Vast since I don't have access to it

Reply:
We added support in Ceph. There is a lot of places this can pop up in PutObject, UploadPart, CompleteMultipartUpload, AWSv2 vs AWSv4 semantics, etc.